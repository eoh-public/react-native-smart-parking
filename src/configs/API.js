import Config from 'react-native-config';

const API_ROOT = Config.API_ROOT;

const API = {
  AUTH: {
    LOGIN_PHONE: API_ROOT + '/accounts/login/',
    LOGIN_FACEBOOK: API_ROOT + '/accounts/login/facebook/',
    REGISTER_PHONE: API_ROOT + '/accounts/register/phone/',
    RESEND_OTP: API_ROOT + '/accounts/register/resend_otp/',
    VERIFY_OTP: API_ROOT + '/accounts/register/verify_otp/',
    LOGIN_SOCIAL_FB: API_ROOT + '/accounts/login/facebook/',
    LOGIN_SOCIAL_GG: API_ROOT + '/accounts/login/google/',
    LOGIN_SOCIAL_AP: API_ROOT + '/accounts/login/apple/',
    FORGOT_PASSWORD: API_ROOT + '/accounts/forgot_password/',
    FORGOT_PASSWORD_VERIFY_OTP:
      API_ROOT + '/accounts/forgot_password/verify_otp/',
    RESET_PASSWORD: API_ROOT + '/accounts/forgot_password/set_password/',
  },
  ACCOUNTS: {
    FEATURES: API_ROOT + '/accounts/features/',
    ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
    REMOVE_CARD: (id) =>
      API_ROOT + `/billing/payments/stripe/remove_cards/${id}/`,
    CHANGE_DEFAULT_CARD: API_ROOT + '/billing/payments/stripe/set_default/',
    LIST_PAYMENT_METHODS: API_ROOT + '/billing/list_payment_methods/',
    CREATE_CARD_TOKEN: 'https://api.stripe.com/v1/tokens',
  },
  UNIT: {
    ADD_USER: API_ROOT + '/property_manager/shared_units/add_by_phone_number/',
    MY_UNITS: API_ROOT + '/property_manager/units/mine/',
    SHARED_UNITS: API_ROOT + '/property_manager/shared_units/',
    UNIT_DETAIL: (id) => API_ROOT + `/property_manager/units/${id}/`,
    UNIT_NEAR_ME: (lat, lon, page) =>
      API_ROOT + `/property_manager/units/near-me/${lat},${lon}/?page=${page}`,
    UNITS_PUBLIC: API_ROOT + '/property_manager/units/public/',
    UNIT_SUMMARY: (id) => API_ROOT + `/property_manager/units/${id}/summary/`,
    UNIT_SUMMARY_DETAIL: (id, id2) =>
      API_ROOT + `/property_manager/units/${id}/summary_detail/${id2}/`,
    FILTER_SHARED_UNITS: (field) =>
      API_ROOT + `/property_manager/shared_units/?ordering=${field}`,
    STAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/star/`,
    UNSTAR_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unstar/`,
    PIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/pin/`,
    UNPIN_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/unpin/`,
    MANAGE_UNIT: (id) => API_ROOT + `/property_manager/units/${id}/`,
  },
  SUB_UNIT: {
    REMOVE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    CREATE_SUB_UNIT: (unitId) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/`,
    MANAGE_SUB_UNIT: (unitId, id) =>
      API_ROOT + `/property_manager/${unitId}/sub_units/${id}/`,
    SENSOR_SCAN: (id) =>
      API_ROOT + `/property_manager/stations/${id}/sensor_scan/`,
  },
  SENSOR: {
    DISPLAY: (id) => API_ROOT + `/property_manager/sensors/${id}/display/`,
    DISPLAY_VALUES: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values/`,
    DISPLAY_VALUES_V2: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/display_values_v2/`,
    DISPLAY_HISTORY: (id, hId) =>
      API_ROOT + `/property_manager/sensors/${id}/display_history/`,
    CAMERA_DISPLAY: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/cameras/`,
    REMOTE_CONTROL_OPTIONS: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/remote_control_options/`,
    QUICK_ACTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/quick_action/`,
    CHECK_CONNECTION: (id) =>
      API_ROOT + `/property_manager/sensors/${id}/check_connection/`,
    UPDATE_SENSOR: (unit_id, station_id, id) =>
      API_ROOT +
      `/property_manager/${unit_id}/sub_units/${station_id}/devices/${id}/`,
  },
  CONFIG: {
    DISPLAY_HISTORY: API_ROOT + '/property_manager/configs/display_history/',
  },
  POWER_CONSUME: {
    DISPLAY_HISTORY:
      API_ROOT + '/property_manager/power_consume/display_history/',
  },
  SHARE: {
    UNITS: API_ROOT + '/property_manager/sharing/units/',
    UNIT_PERMISSIONS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/permissions/`,
    UNITS_MEMBERS: (id) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/`,
    UNITS_MEMBER_DETAIL: (id, shareId) =>
      API_ROOT + `/property_manager/sharing/units/${id}/members/${shareId}/`,
    SHARE: API_ROOT + '/property_manager/sharing/share/',
  },
  EXTERNAL: {
    WEATHER: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/weather?lat=${location.lat}&lon=${location.lon}` +
      `&appid=${weatherMapId}`,
    UV_INDEX: (location, weatherMapId) =>
      `https://api.openweathermap.org/data/2.5/uvi?appid=${weatherMapId}&lat=${location.lat}&lon=${location.lon}`,
    AQI: (location, weatherMapToken) =>
      `https://api.waqi.info/feed/geo:${location.lat};${location.lon}/?token=${weatherMapToken}`,
    GOOGLE_MAP: {
      AUTO_COMPLETE:
        'https://maps.googleapis.com/maps/api/place/autocomplete/json',
      GET_LAT_LNG_BY_PLACE_ID:
        'https://maps.googleapis.com/maps/api/place/details/json',
    },
  },
  PARKING: {
    NEARBY: API_ROOT + '/smart_parking/parking_areas/near_me/',
    NEAREST:
      API_ROOT + '/smart_parking/parking_areas/nearest_available_parking/',
    SAVED_LIST: API_ROOT + '/smart_parking/parking_areas/saved_list/',
    SAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/save/`,
    UNSAVE: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/unsave/`,
    DETAIL: (id) => API_ROOT + `/smart_parking/parking_areas/${id}/`,
    GET_BOOKING_PRICE: (id) =>
      API_ROOT + `/smart_parking/parking_areas/${id}/get_price/`,
    AVAILABLE_TIME_SLOTS: (id) =>
      API_ROOT + `/smart_parking/parkings/${id}/available-time-slots/`,
    PAYMENT_SUCCESS: (id) => API_ROOT + `/smart_parking/payment_success/${id}/`,
    CHECK_CAR_PARKED: API_ROOT + '/smart_parking/spots/check_car_parked/',
    PARKING_INFO: API_ROOT + '/smart_parking/spots/parking_info/',
  },
  CAR: {
    MY_CARS: API_ROOT + '/smart_parking/cars/',
    CHECK_CARS_INFO: API_ROOT + '/smart_parking/cars/check_information/',
    REMOVE_CAR: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE: (id) => API_ROOT + `/smart_parking/cars/${id}/`,
    UPDATE_DEFAULT: (id) =>
      API_ROOT + `/smart_parking/cars/${id}/update_default/`,
  },
  BOOKING: {
    CREATE: API_ROOT + '/smart_parking/bookings/',
    DETAIL: (id) => API_ROOT + `/smart_parking/bookings/${id}/`,
    ACTIVE_SESSION: API_ROOT + '/smart_parking/bookings/active_session/',
    HISTORY: (page) =>
      API_ROOT + `/smart_parking/bookings/history/?page=${page}`,
    SCAN_TO_BOOK: API_ROOT + '/smart_parking/bookings/scan_to_book/',
    SCAN_TO_CONFIRM: API_ROOT + '/smart_parking/bookings/scan_to_confirm/',
    EXTEND_INFO: (id) =>
      API_ROOT + `/smart_parking/bookings/${id}/extend_info/`,
    EXTEND: (id) => API_ROOT + `/smart_parking/bookings/${id}/extend/`,
    CANCEL: (id) => API_ROOT + `/smart_parking/bookings/${id}/cancel/`,
    STOP: (id) => API_ROOT + `/smart_parking/bookings/${id}/stop/`,
  },
  EMERGENCY: {
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
  },
  BILLING: {
    LIST_PAYMENT_METHODS_BY_COUNTRY: (code) =>
      API_ROOT + `/billing/list-methods/${code}/`,
    PAYMENT: {
      STRIPE: {
        PROCESS: (id) => API_ROOT + `/billing/payments/stripe/process/${id}/`,
        ADD_CARD: API_ROOT + '/billing/payments/stripe/add_card/',
        CREATE_PAYMENT_INTENT: (id) =>
          API_ROOT + `/billing/payments/stripe/sca/create-payment/${id}/`,
        PAYMENT_INTENT_SUCCESS: (intent_id) =>
          API_ROOT +
          `/billing/payments/stripe/sca/payment_success/${intent_id}/`,
      },
    },
    DEFAULT_PAYMENT_METHODS: API_ROOT + '/billing/default_payment_method/',
  },
  NOTIFICATION: {
    REGISTER_SIGNAL_ID: API_ROOT + '/accounts/register-signal-id/',
    LIST_ALL_NOTIFICATIONS: (page, type) =>
      API_ROOT + `/notifications/notifications/?page=${page}&type=${type}`,
    SET_READ: (id) => API_ROOT + `/notifications/notifications/${id}/set_read/`,
    NUMBER: API_ROOT + '/notifications/notifications/number/',
  },
  EMERGENCY_BUTTON: {
    CREATE_CONTACT: API_ROOT + '/emergency_button/contacts/',
    CONTACTS: API_ROOT + '/emergency_button/contacts/',
    REMOVE_CONTACTS: (id) => API_ROOT + `/emergency_button/contacts/${id}/`,
    SEND_ALERT: API_ROOT + '/emergency_button/events/',
    NOTIFY_SMS: (id) => API_ROOT + `/emergency_button/events/${id}/notify_sms/`,
    FASLE_ALARM: (id) =>
      API_ROOT + `/emergency_button/events/${id}/false_alarm/`,
    RESOLVE: (id) => API_ROOT + `/emergency_button/events/${id}/resolve/`,
  },
  IOT: {
    CHIP_MANAGER: {
      WATCH_CONFIGS: API_ROOT + '/chip_manager/watch_configs/',
      PUSHER_AUTH: API_ROOT + '/chip_manager/pusher/auth/',
    },
  },
};

export default API;
