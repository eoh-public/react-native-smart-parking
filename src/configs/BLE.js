export default {
  BLE_REMOTE_SERVICE_UUID: '4fafc201-1fb5-459e-8fcc-c5c9c331914b',
  BLE_REMOTE_CHARACTERISTIC_UUID: 'beb5483e-36e1-4688-b7f5-ea07361b26a8',
};
