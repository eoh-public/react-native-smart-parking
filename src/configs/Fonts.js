export const Fonts = {
  Regular: 'SFProDisplay-Regular',
};
