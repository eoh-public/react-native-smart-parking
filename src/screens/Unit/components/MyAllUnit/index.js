import React, { useRef } from 'react';
import { StyleSheet, SafeAreaView, Animated, ScrollView } from 'react-native';

import Header from '../Header';
import ListMyAllUnit from '../ListMyAllUnit';
import { Colors } from '../../../../configs';

const AnimatedScrollView = Animated.createAnimatedComponent(ScrollView);

const MyAllUnit = ({ route, navigation }) => {
  const animatedScrollYValue = useRef(new Animated.Value(0)).current;
  const unitItems = route.params.myUnits;

  return (
    <SafeAreaView style={[styles.wrap, styles.backgroundWhite]}>
      <Header dark goBack={() => navigation.goBack()} />
      <AnimatedScrollView
        scrollEventThrottle={1}
        onScroll={Animated.event(
          [{ nativeEvent: { contentOffset: { y: animatedScrollYValue } } }],
          {
            useNativeDriver: true,
          }
        )}
        contentContainerStyle={styles.wrap}
      >
        <ListMyAllUnit unitItems={unitItems} />
      </AnimatedScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  wrap: {
    flex: 1,
  },
  backgroundWhite: {
    backgroundColor: Colors.White,
  },
});

export default MyAllUnit;
