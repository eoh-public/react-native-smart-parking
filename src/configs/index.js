import { SPConfig, initSPConfig } from './SPConfig';
import API from './API';
import { Colors } from './Colors';
import { Constants } from './Constants';
import Theme from './Theme';
import Images from './Images';
import Styles from './Styles';
import BLE from './BLE';
import Device from './Device';
import { Fonts } from './Fonts';
export {
  API,
  BLE,
  Colors,
  Constants,
  Theme,
  Styles,
  Images,
  Device,
  Fonts,
  SPConfig,
  initSPConfig,
};
