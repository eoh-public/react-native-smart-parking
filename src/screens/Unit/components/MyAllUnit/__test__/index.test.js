import React from 'react';
import { TouchableOpacity } from 'react-native';
import { act, create } from 'react-test-renderer';
import MyAllUnit from '..';
import { TESTID } from '../../../../../configs/Constants';

describe('Test MyAllUnit', () => {
  let data;
  const mockGoBack = jest.fn();

  beforeEach(() => {
    data = {
      route: {
        params: {
          myUnits: undefined,
        },
      },
      navigation: {
        goBack: mockGoBack,
      },
    };
  });
  let tree;

  test('render MyAllUnit', async () => {
    act(() => {
      tree = create(<MyAllUnit {...data} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('goback', async () => {
    act(() => {
      tree = create(<MyAllUnit {...data} />);
    });
    const instance = tree.root;
    const backDefaults = instance.findAll(
      (el) =>
        (el.props.testID =
          TESTID.BACK_DEFAULT_TOUCH && el.type === TouchableOpacity)
    );
    act(() => {
      backDefaults[1].props.onPress();
    });
    expect(mockGoBack).toHaveBeenCalledTimes(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
