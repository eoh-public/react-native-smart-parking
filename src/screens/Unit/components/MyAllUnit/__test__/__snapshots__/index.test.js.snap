// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyAllUnit goback 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 10,
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderBottomWidth": 1,
                  "borderColor": "#E8E8E8",
                  "borderTopLeftRadius": 20,
                  "borderTopRightRadius": 20,
                  "justifyContent": "center",
                  "padding": 16,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SFProDisplay-Regular",
                      "fontSize": undefined,
                      "lineHeight": undefined,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "color": "#262626",
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                  ]
                }
              >
                Thêm
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                  "padding": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": "33.33%",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <SvgMock
                      height={43}
                      width={43}
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "SFProDisplay-Regular",
                          "fontSize": undefined,
                          "lineHeight": undefined,
                          "textAlign": "left",
                          "textDecorationLine": "none",
                        },
                        Object {},
                      ]
                    }
                    styles={
                      Object {
                        "color": "#595959",
                        "fontSize": 14,
                        "lineHeight": 30,
                      }
                    }
                  >
                    Thành viên
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <Modal
      hardwareAccelerated={true}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      onShow={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
        ]
      }
      transparent={true}
      visible={false}
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
            Object {
              "minWidth": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingLeft": 12,
              "paddingVertical": 0,
            }
          }
          testID="BACK_DEFAULT_TOUCH"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "antoutline",
                  "fontSize": 27,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "fontSize": 16,
              },
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
            Object {
              "minWidth": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingLeft": 5,
              "paddingRight": 15,
              "paddingTop": 2,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "antoutline",
                    "fontSize": 32,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "antoutline",
                    "fontSize": 30,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    onScroll={[Function]}
    scrollEventThrottle={1}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "paddingTop": 20,
            }
          }
          onScroll={[Function]}
          scrollEventThrottle={1}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Test MyAllUnit render MyAllUnit 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "width": 750,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 10,
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderBottomWidth": 1,
                  "borderColor": "#E8E8E8",
                  "borderTopLeftRadius": 20,
                  "borderTopRightRadius": 20,
                  "justifyContent": "center",
                  "padding": 16,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SFProDisplay-Regular",
                      "fontSize": undefined,
                      "lineHeight": undefined,
                      "textAlign": "left",
                      "textDecorationLine": "none",
                    },
                    Object {
                      "color": "#262626",
                      "fontSize": 16,
                      "lineHeight": 24,
                    },
                  ]
                }
              >
                Thêm
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "center",
                  "padding": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": "33.33%",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <SvgMock
                      height={43}
                      width={43}
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "SFProDisplay-Regular",
                          "fontSize": undefined,
                          "lineHeight": undefined,
                          "textAlign": "left",
                          "textDecorationLine": "none",
                        },
                        Object {},
                      ]
                    }
                    styles={
                      Object {
                        "color": "#595959",
                        "fontSize": 14,
                        "lineHeight": 30,
                      }
                    }
                  >
                    Thành viên
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <Modal
      hardwareAccelerated={true}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      onShow={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
        ]
      }
      transparent={true}
      visible={false}
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
            Object {
              "minWidth": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "paddingLeft": 12,
              "paddingVertical": 0,
            }
          }
          testID="BACK_DEFAULT_TOUCH"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "antoutline",
                  "fontSize": 27,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "fontSize": 16,
              },
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
            Object {
              "minWidth": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingLeft": 5,
              "paddingRight": 15,
              "paddingTop": 2,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "antoutline",
                    "fontSize": 32,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "antoutline",
                    "fontSize": 30,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  undefined,
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    onScroll={[Function]}
    scrollEventThrottle={1}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "paddingTop": 20,
            }
          }
          onScroll={[Function]}
          scrollEventThrottle={1}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
